osu.Game.Rulesets.Taiko/Skinning/Legacy/TaikoLegacySkinTransformer.cs

```diff
@@ -159,31 +156,5 @@ private string getHitName(TaikoSkinComponents component)
 
             throw new ArgumentOutOfRangeException(nameof(component), $"Invalid component type: {component}");
         }
-
-        public override ISample? GetSample(ISampleInfo sampleInfo)
-        {
-            if (sampleInfo is HitSampleInfo hitSampleInfo)
-                return base.GetSample(new LegacyTaikoSampleInfo(hitSampleInfo));
-
-            return base.GetSample(sampleInfo);
```

Comment:
The reasoning for the change in lookup precedence is the removal of this.

Because this code is in the transformer, each call to `base.GetSample` was doing a full hierarchical lookup. First with `taiko-` prefixes and *then* falling back to non-`taiko-` samples.

With the new code, this has changed to do lookups for `taiko-` and then non-`taiko-` on a *per skin* basis, which matches watch @nekodex reported seeing changed.

I haven't yet looked into a resolution for this, but I'd guess we need to bring this code back here, and apply the `taiko-` prefix locally rather than in `TaikoHitSampleInfo`.

Reply:
For sure. I think this is tested in osu! ruleset, but not taiko.

Reply:
I've started to look into a fix for this, but it looks like it will be a complete revert of `TaikoHitSampleInfo`. It doesn't need to exist and is only adding complexity.

Before I address this though, I'm waiting for feedback from @nekodex regarding sample banks.

In stable skins, taiko has normal/soft/drum banks, but in this new implementation only one bank exists. But then there's a new "velocity" suffix which is being determined from sample control point volume. This all feels like an oversight to me, so I want to know what the path forward is here before making changes (as they may change the structure of the sample filenames).